pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1116,6 +1198,69 @@ func lookupDomainConfig(ctx *domainContext, key string) *types.DomainConfig {
 	return &config
 }
 
+func setCgroupCpuset(config *types.DomainConfig) error {
+	cgroupName := filepath.Join(containerd.GetServicesNamespace(), config.GetTaskName())
+	cgroupPath := cgroups.StaticPath(cgroupName)
+	controller, err := cgroups.Load(cgroups.V1, cgroupPath)
+	if err != nil {
+		// It's still not an error, since the path may still not exist
+		log.Warnf("Failed to find cgroups directory for %s", config.DisplayName)
+		return nil
+	}
+	err = controller.Update(&specs.LinuxResources{CPU: &specs.LinuxCPU{Cpus: config.VmConfig.CPUs}})
+	if err != nil {
+		log.Warnf("Failed to update CPU set for %s", config.DisplayName)
+		return err
+	}
+	return nil
+}
+
+func constructNonPinnedCpumaskString(ctx *domainContext) string {
+	result := ""
+	for _, cpu := range ctx.cpuAllocator.GetAllFree() {
+		addToMask(cpu, &result)
+	}
+	return result
+}
+
+func addToMask(cpu int, s *string) {
+	if s == nil {
+		return
+	}
+	if *s == "" {
+		*s = fmt.Sprintf("%d", cpu)
+	} else {
+		*s = fmt.Sprintf("%s,%d", *s, cpu)
+	}
+}
+
+func updateNonPinnedCPUs(ctx *domainContext, config *types.DomainConfig) error {
+	config.VmConfig.CPUs = constructNonPinnedCpumaskString(ctx)
```

Comment:
Is it possible to not modify DomainConfig in domainmgr? Domainmgr has subscription for this type and the expectation is to get DomainConfig as is from another agent and react using DomainStatus.

Reply:
I think it's possible. And I understand the point. It's valid. I've checked the code, and it really looks like we change the Config here only.

But I would prefer to get this kind of comment first, not after I already reimplement the entire PR.
@eriknordmark, should I redo it once more now?

Reply:
It makes sense to put the update in DomainStatus (and leave DomainConfig as it was received from zedmanager).
It probably is easiest to put the CPUs field in DomainStatus (that should be easier than introducing some new VmStatus which has a superset of the fields in VmConfig)
I think such a change is fairly localized.

Reply:
I'll try to do it... Not sure it work, but I'll try

Reply:
To do it, we have to modify the `UpdateFromDomain` method of the OCI interface:
https://github.com/lf-edge/eve/blob/1844bc4b5a825fd33a06b7630136a6cbc6e3797d/pkg/pillar/containerd/oci.go#L64
We need to add DomainSataus as a parameter there. Will it work?

Reply:
Not sure, as I can see from the comment, it is about DomainConfig, something that we explicitly receive from the config. Probably we can expand adjustment function we have [here](https://github.com/lf-edge/eve/blob/1844bc4b5a825fd33a06b7630136a6cbc6e3797d/pkg/pillar/hypervisor/kvm.go#L511) or implement another one.

Reply:
I still do not see any reason not to add Status to `UpdateFromDomain`.
Or even like this: I do not even see the reason for `UpdateFromDomain` to exist if we have finer-grained functions like `UpdateVifList`, `UpdateFromVolume` and `AdjustMemLimit`. Btw, why the last one does not follow the `UpdateFrom*` pattern?
But actually, I don't care, I can add a new function.

Reply:
Ah, anyway, adding Status to UpdateFromDomain does not help.

Reply:
Argh... I do not see a good and easy solution to the problem. Whenever I try to move the actual CPUs data to DomainStatus, all the tests start failing :(

Reply:
The problem is that in `kvm.go` DomainConfig is used to create the `xen*.cfg` file, not DomainSatus.

Reply:
Ok, I use a local copy of DomainConfig to create a `xen*.cfg` file. Works fine now.

Reply:
Should we resolve this thread ?

Reply:
Yes, thank you. We still modify DomainConfig outside of zedmanager, but we may optimize it later.